changelog.d/8726.bugfix

```diff
@@ -0,0 +1 @@
+Fix bug where Synapse would not recover after losing connection to the database.
```

Comment:
Assuming https://github.com/matrix-org/synapse/pull/8456 was what led to the breakage(?), then the issue was introduced in v1.21.0.

Reply:
I really don't see why this would be more of an issue in v1.21 TBH, so I'm slightly hesitant to say this is fixing a regression :shrug: